jupyterhub/templates/proxy/deployment.yaml

```diff
@@ -24,8 +24,9 @@ spec:
         {{- . | toYaml | trimSuffix "\n" | nindent 8 }}
         {{- end }}
       annotations:
-        # This lets us autorestart when the secret changes!
-        checksum/hub-secret: {{ include (print $.Template.BasePath "/hub/secret.yaml") . | sha256sum }}
+        # This restarts the pod when the proxy secret changes.
+        # The proxy secret is *only* present when doing manual HTTPS
+        # certs, so restarting is required to get those changes.
         checksum/proxy-secret: {{ include (print $.Template.BasePath "/proxy/secret.yaml") . | sha256sum }}
```

Comment:
I'm not sure about this comment, especially `so restarting is required to get those changes.`

Changes to `proxy.https.type=manual` and `proxy.https.manual.cert|key` will trigger a restart through this checksum by rendering the proxy-secret.

What do you think about removing the comment and updating the checksum annotation name to `checksum/manual-https-certs` or similar?

Reply:
Since we want the value to come from a secret, is it sensible/reliable to copy the value to the proxy secret given the current existing-secret / lookup / random-if-not-set behavior?

If so, then:

1. the env value comes from a secret (required, I think)
2. proxy-secret always exists
3. proxy-secret hash always makes sense, following existing practice of 'any change here means it should restart' like we do for the hub

Reply:
I don't see the mechanism to help us copy the value reliably.

If we actually render secret.yaml multiple times, like we actually do when setting the checksum annotation, we end up with multiple random seeds anyhow. The crux is that we don't have a way to memoize the random output which we want to get from multiple Helm templates during a chart rendering.

Reply:
If we can memoize the random output, we can avoid any unwanted restarts.

If we would duplicate the CHP auth token to the proxy k8s Secret otherwise only used for manually provided http certs, we would still be forced to use the same Helm template helper which relies on the: explicit config, the hub k8s Secret value, or a separate random seed because we haven't memoized the helper.

My suggestion:
1. We accept _one_ unwanted restart following the initial CHP auth token generation. 
   Note that this is caused by syncing with whatever was generated separately from either the proxy k8s Secret or the proxy k8s Deployment annotation with what was generated for the hub k8s Secret.
2. We use an checksum annotation instead of creating a k8s Secret for the proxy pod. If we would, we would _need to not mount it_, as then we would have a mismatch when the password is generated causing communication failure.